esi_ui/static/dashboard/project/esi/actions.service.js

```diff
@@ -0,0 +1,79 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('horizon.dashboard.project.esi')
+    .factory('actionsService', actionsService);
+
+    actionsService.$inject = [
+      '$q',
+      'nodesService',
+    ];
+
+    function actionsService($q, nodesService) {
+      var leftSplitButtonAction = {
+        type: 'row',
+        service: {
+          allowed: function(node) {
+            return $q.resolve(true);
+          },
+          perform: function(node) {
+            console.log('actionsService: leftSplitButtonAction ', node);
+          }
+        },
+        template: {
+          text: 'Edit',
+          actionClasses: 'split-caret dropdown-toggle btn btn-default',
+        }
+      };
+
+      var setPowerOnAction = {
+        type: 'row',
+        service: {
+          allowed: function(node) {
+            return $q.resolve(true);
+          },
+          perform: function(node) {
+            return setPowerState(node, 'power on');
+          }
+        },
+        template: {
+          text: 'Power On',
+        }
+      };
+
+      var setPowerOffAction = {
+        type: 'row',
+        service: {
+          allowed: function(node) {
+            return $q.resolve(true);
+          },
+          perform: function(node) {
+            return setPowerState(node, 'power off');
+          }
+        },
+        template: {
+          text: 'Power Off',
+        }
+      };
+
+      var actions = [
+        leftSplitButtonAction,
+        setPowerOnAction,
+        setPowerOffAction,
+      ];
+      return {
+        getActions: function() {
+          return actions;
```

Comment:
This is definitely possible, but I think this would be very hard to implement with the structure I currently have. The itemActions array is a list of action buttons that show up on all rows in the list, not on an individual level, with the row node passed into the callback function. For now, I already implemented a mechanism that gives an error if the user requests to change the state to what it currently is.

Reply:
That's reasonable, and an error message does address my concern - thanks!